packages/db-kit/src/utils/prepareProjectInfo.ts

```diff
@@ -0,0 +1,75 @@
+import { Resources } from "@truffle/db";
+import type { ProjectInfo } from "@truffle/decoder";
+
+export interface PrepareProjectInfoOptions {
+  compilations: Resources.Resource<"compilations">[];
+}
+
+export async function prepareProjectInfo({
```

Comment:
This is neat! So the code here makes the compilation output decoder-friendly? What is the plan for this eventually? When DB replaces artifacts will the decoder be rewritten to expect things in the DB structure?

Reply:
The decoder defines its own format, partially because that's just how we did it, but also because it has its own requirements. We might be able to update parts of the ProjectInfo type to align with DB, but it's not really decided yet. I've also got it in mind that maybe the better alternative is to have the decoder query @truffle/db directly (as an alternative to passing in this information upfront)